extra/3-magic-8-ball.js

```diff
@@ -58,10 +88,25 @@ function shakeBall() {
 
   This function should expect to be called with any value which was returned by the shakeBall function.
 */
-function checkAnswer(answer) {
-  //Write your code in here
-}
 
+ function checkAnswer(answer) {
+   //Write your code in here
+   let ballPredict = fixAnswer.indexOf(answer);
+   if (ballPredict <= 4) {
+     answer = "very positive";
+   } else if (ballPredict <= 9) {
+     answer = "positive";
+   } else if (ballPredict <= 14) {
+     answer = "negative";
+   } else if (ballPredict <= 19) {
+     answer = "very negative";
+   } else answer = "Something broken";
```

Comment:
Hi Leila, you've pretty much hit the nail on the head here 👍 

I would just be careful at the end with your `} else answer = "Something broken";` as this syntax isn't quite correct. I also notice you define answer then return it at the end, another way of doing this is just to return in each conditional block. For example:

```javascript
function checkAnswer(answer) {
  const ballPredict = fixAnswer.indexOf(answer);
  
  if (ballPredict <= 4) {
    return "very positive";
  } else if (ballPredict <= 9) {
    return "positive";
  } else if (ballPredict <= 14) {
    return "negative";
  } else if (ballPredict <= 19) {
    return "very negative";
  }
  
  return "Something broken";
}
```